lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -124,16 +129,224 @@ export class BrowserCacheManager extends CacheManager {
+     */
+    async migrateExistingCache(correlationId: string): Promise<void> {
+        const accountKeys0 = getAccountKeys(this.browserStorage, 0);
+        const tokenKeys0 = getTokenKeys(this.clientId, this.browserStorage, 0);
+        this.performanceClient.addFields(
+            {
+                oldAccountCount: accountKeys0.length,
+                oldAccessCount: tokenKeys0.accessToken.length,
+                oldIdCount: tokenKeys0.idToken.length,
+                oldRefreshCount: tokenKeys0.refreshToken.length,
+            },
+            correlationId
+        );
+
+        const accountKeys1 = getAccountKeys(this.browserStorage, 1);
+        const tokenKeys1 = getTokenKeys(this.clientId, this.browserStorage, 1);
+        this.performanceClient.addFields(
+            {
+                currAccountCount: accountKeys1.length,
+                currAccessCount: tokenKeys1.accessToken.length,
+                currIdCount: tokenKeys1.idToken.length,
+                currRefreshCount: tokenKeys1.refreshToken.length,
+            },
+            correlationId
+        );
+
+        await Promise.all([
+            this.updateV0ToCurrent(
+                CacheKeys.ACCOUNT_SCHEMA_VERSION,
+                accountKeys0,
+                accountKeys1,
+                correlationId
+            ),
+            this.updateV0ToCurrent(
+                CacheKeys.CREDENTIAL_SCHEMA_VERSION,
+                tokenKeys0.idToken,
+                tokenKeys1.idToken,
+                correlationId
+            ),
+            this.updateV0ToCurrent(
+                CacheKeys.CREDENTIAL_SCHEMA_VERSION,
+                tokenKeys0.accessToken,
+                tokenKeys1.accessToken,
+                correlationId
+            ),
+            this.updateV0ToCurrent(
+                CacheKeys.CREDENTIAL_SCHEMA_VERSION,
+                tokenKeys0.refreshToken,
+                tokenKeys1.refreshToken,
+                correlationId
+            ),
+        ]);
+
+        if (accountKeys0.length > 0) {
+            this.browserStorage.setItem(
+                CacheKeys.getAccountKeysCacheKey(0),
+                JSON.stringify(accountKeys0)
+            );
+        } else {
+            this.browserStorage.removeItem(CacheKeys.getAccountKeysCacheKey(0));
+        }
+
+        if (accountKeys1.length > 0) {
+            this.browserStorage.setItem(
+                CacheKeys.getAccountKeysCacheKey(1),
+                JSON.stringify(accountKeys1)
+            );
+        } else {
+            this.browserStorage.removeItem(CacheKeys.getAccountKeysCacheKey(1));
+        }
+
+        this.setTokenKeys(tokenKeys0, correlationId, 0);
+        this.setTokenKeys(tokenKeys1, correlationId, 1);
+    }
+
+    async updateV0ToCurrent(
+        currentSchema: number,
+        v0Keys: Array<string>,
+        v1Keys: Array<string>,
+        correlationId: string
+    ): Promise<void[]> {
+        const upgradePromises: Array<Promise<void>> = [];
```

Comment:
I get the idea, but this function is very hard to read. It would be great if you could decompose it later.